### PR TITLE
feat: Add ability to delete product

### LINKED ISSUE #122: Add Option to Delete a Product
Need to be able to delete product from product details page.

### PR DESCRIPTION
**Pull request by DwangDistracted**
This feature allows you to delete a product. This can be done from the product details page and results in the user being redirected to the store page.

Misc changes are to make the product creation screen able to handle small viewports by using relative % based margins.

Closes #122

### CODE REVIEW

File: shopify-client/src/views/ProductCreate.vue [hunk 1]
```diff
@@ -63,7 +63,7 @@ export default {
 }
 
 .center {
-  margin-left: 500px;
-  margin-right: 500px;
+  margin-left: 10%;
```
Comment (mpfan) [inline shopify-client/src/views/ProductCreate.vue:66]: Just a comment but % should not be used for margins. Different breakpoints should be used to specify the margins.

**Review by mpfan — APPROVED** [review]

**Review by dershao — APPROVED** [review]

File: src/main/java/com/shopify/minishopify/model/Shop.java [hunk 2]
```diff
@@ -141,6 +141,11 @@ public void addProduct(Product product) {
         products.add(product);
     }
 
+    public void removeProduct(Product product) {
+        product.setShop(null);
```
[the remark this answers is not in the document]
Comment (DwangDistracted) [reply src/main/java/com/shopify/minishopify/model/Shop.java:145]: yeah. Just needed to sever the connection between product and shop.

**Review by Weebian — COMMENTED** [review]
When I add the product to the cart then delete it, it is successfully removes it from the cart. 
But the header still show the cart size as 1.
image

**Comment by DwangDistracted** [comment]
> When I add the product to the cart then delete it, it is successfully removes it from the cart.
> But the header still show the cart size as 1.
> Not sure if you wanna get it fixed this issue
> image

Should be fixed.

**Review by mpfan — APPROVED** [review]

**Review by dershao — APPROVED** [review]

**Pull request merged**